EateryModel/Sources/EateryModel/Networking/EateryAPI.swift

```diff
@@ -31,5 +31,28 @@ public struct EateryAPI {
         return schemaApiResponse.data.map(SchemaToModel.convert)
     }
 
+    public func reportError(eateryId: Int64? = nil, type: String, content: String) async {
```

Comment:
To use async you'll also need to rewrite this to use URLSession's async methods. See: 
- https://developer.apple.com/documentation/foundation/urlsession/3767353-data
- https://github.com/cuappdev/eatery-blue-ios/blob/266d7ceeccd4d53498309a0cb8dbeed0f6081468/EateryModel/Sources/EateryModel/Networking/EateryAPI.swift#L23

Reply:
I'm actually not super sure how to do this, I keep getting a ```AppDelegate.main()``` error

This is the method I have so far:
```
  public func reportError(eateryId: Int64? = nil, type: String, content: String) async {
      let data = try? JSONEncoder.init().encode(
          Schema.ErrorReport(
              eateryId: eateryId,
              type: type,
              content: content)
      )
      let body = try? JSONSerialization.data(withJSONObject: data, options: [])

      var request = URLRequest(url: url)
      request.httpBody = body
      request.httpMethod = "POST"

      do {
          let (_, response) = try await URLSession.shared.data(for: request)
          let httpResponse = response as? HTTPURLResponse
          switch httpResponse?.statusCode {
          case 200: logger.info("Successfully reported Eatery Blue issue")
          default: throw EateryAPIError.apiResponseError("Unable to submit report")
          }
      } catch {
          logger.error("Unknown error submitting report")
      }

  }
```